client/app/scripts/superdesk/directives/sdUserAvatar.js

```diff
@@ -14,16 +14,21 @@ define(['angular'], function (angular) {
                         element.hide();
                     });
 
-                    if (scope.src) {
-                        element.attr('src', scope.src).show();
-                        figure.addClass('no-bg');
-                    } else if (scope.initials) {
-                        var initials = scope.initials.replace(/\W*(\w)\w*/g, '$1').toUpperCase();
-                        element.hide().parent().html(initials);
-                        figure.addClass('initials');
-                    } else {
-                        element.hide();
-                        figure.removeClass('no-bg');
+                    scope.$watch('src', initAvatar);
```

Comment:
Good point! I will change this to pass the user to directive and there to check if it needs to show avatar or initials!